packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts

```diff
@@ -138,6 +140,10 @@ export const useNotifiSubscribe: ({
 
       setIsSmsConfirmed(isPhoneNumberConfirmed);
 
+      if (!isPhoneNumberConfirmed) {
+        setPhoneNumberErrorMessage('Invalid number');
+      }
```

Comment:
seeing this makes me feel like there's a lot of undefined state that's giong to be tough to maintain. How hard would it be to re-design the `confirmed` state to use the new "destination error" state?

spitballing off top of my head, but maybe something like
```
type RecoverableError = Readonly<{
  type: 'recoverableError';
  label: string;
  onClick: () => void;
}>;

type UnrecoverableError = Readonly<{
  type: 'unrecoverableError';
  label: string;
  tooltip?: string;
}>;

type DestinationError = UnrecoverableError | RecoverableError;
type DestinationErrors = Record<FormField, DestinationError>;
```

Reply:
will take a look!